### PR TITLE
List mobile clients on mobile services

### PR DESCRIPTION
**Pull request by sedroche**
Ticket: https://issues.jboss.org/browse/AEROGEAR-2156

Initial displaying of clients within the service instance of the overview page.

- Display clients only on mobile services
- Only show clients that are not excluded from that service

apps-serviceinstance

**Note:** Cordova icon included in this [PR](https://github.com/openshift/openshift-logos-icon/pull/11) so wont be available yet if testing

### CODE REVIEW

**Comment by sedroche** [comment]
@witmicko @JameelB @vchepeli @dimitraz

File: app/scripts/directives/overview/mobileClientRow.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    _.extend(row, ListRowUtils.ui);
+
+    row.bundleDisplay = "bundle id: " + row.apiObject.spec.appIdentifier;
+    switch (row.apiObject.spec.clientType) {
+      case 'android':
+        row.installType = 'gradle';
+        row.bundleDisplay = "package name: " + row.apiObject.spec.appIdentifier;
+        break;
+      case 'iOS':
+        row.installType = 'cocoapods';
```
Comment (JameelB) [inline app/scripts/directives/overview/mobileClientRow.js]: The iOS and cordova apps now have appdentifiers as well. Should `row.bundleDisplay` be added here too?

**Comment by witmicko** [comment]
@sedroche should it mimic this view https://www.dropbox.com/s/zjb4h64jo279654/Screenshot%202018-02-02%2011.29.07.png?dl=0 ?

File: app/views/overview/_service-instance-row.html [hunk 2]
```diff
@@ -152,6 +152,18 @@ <h3>
...
+          <div class="col-sm-6">
+            <div ng-repeat="mobileClient in row.filteredClients" class="col-sm-12 col-md-6 mobile-clients">
+              <span class="fa-2x {{mobileClient.metadata.annotations.icon}}"></span>
+              <a ng-href="{{mobileClient | navigateResourceURL}}">{{mobileClient.metadata.name}}</a>
+            </div>
+          </div>
+        </div>
+        <div class="row">
+          <div class="col-sm-12">
             <div ng-if="row.serviceClass.spec.externalMetadata.documentationUrl || row.serviceClass.spec.externalMetadata.supportUrl">
```
Comment (witmicko) [inline app/views/overview/_service-instance-row.html]: can you please get it closer to the wirefarme https://www.dropbox.com/s/zjb4h64jo279654/Screenshot%202018-02-02%2011.29.07.png?dl=0
```
 icon   {{client.state.name}}
 icon   clientType/id or some other info
```

File: app/scripts/directives/overview/serviceInstanceRow.js [hunk 3]
```diff
@@ -32,12 +33,13 @@
     var serviceInstanceFailedMessage = $filter('serviceInstanceFailedMessage');
     var truncate = $filter('truncate');
 
-    _.extend(row, ListRowUtils.ui);
+    _.extend(row, ListRowUtils.ui)
```
Comment (maleck13) [inline app/scripts/directives/overview/serviceInstanceRow.js]: should this semi colon be missing?

Comment (sedroche) [reply app/scripts/directives/overview/serviceInstanceRow.js]: yeah not sure why that crept in. will update

File: app/views/overview/_service-instance-row.html (hunk 2 above)
Comment (sedroche) [reply app/views/overview/_service-instance-row.html]: This is the new look @witmicko 

new-client-layout

wyt?

Comment (witmicko) [reply app/views/overview/_service-instance-row.html]: this is perfect @sedroche

Comment (sedroche) [reply app/views/overview/_service-instance-row.html]: :+1:

Comment (dimitraz) [reply app/views/overview/_service-instance-row.html]: @sedroche is it possible to make the client names bold?

File: app/styles/_overview.less [hunk 4]
```diff
@@ -455,16 +455,49 @@
...
+          .client-type {
+            padding-left: 10px;
+            color: #757575;
+            font-size: 11px;
+            line-height: 11px;
+          }
+        }
+
+        a {
+          padding-left: 1em;
```
Comment (dimitraz) [inline app/styles/_overview.less]: @sedroche i know this is very nit picky, but would it be possible to change this to `10px` as well, just to left align the 2 items?

Comment (dimitraz) [reply app/styles/_overview.less]: image

Comment (sedroche) [reply app/styles/_overview.less]: :+1:

File: app/scripts/directives/overview/mobileClientRow.js (hunk 1 above)
Comment (maleck13) [reply app/scripts/directives/overview/mobileClientRow.js]: It has a default value at line 32

File: app/scripts/directives/overview/serviceInstanceRow.js [hunk 5]
```diff
@@ -25,7 +28,9 @@
                               AuthorizationService,
                               BindingService,
                               ListRowUtils,
-                              ServiceInstancesService) {
+                              ServiceInstancesService,
```
Comment (maleck13) [inline app/scripts/directives/overview/serviceInstanceRow.js]: They have a standard of keeping these in alphabetical order same with the above list

**Pull request closed without merging**